File: Miku-AI/src/lib/quietHours.ts
import { QUIET_HOURS_END_HOUR, QUIET_HOURS_START_HOUR } from "../config/constants";
import { isStreamModeActive } from "./streamMode";
import { isGameModeActive } from "./gameMode";

// Idea #20: franja horaria en la que ningún aviso automático (correo,
// Calendar, pendientes) habla solo -- se acumulan y se leen cuando termina.
// Soporta que la franja cruce la medianoche (ej. 23 a 8) además del caso
// simple (0 a 7) que se usa por defecto.
export function isQuietHours(now: Date = new Date()): boolean {
  const hour = now.getHours();
  if (QUIET_HOURS_START_HOUR === QUIET_HOURS_END_HOUR) return false;
  if (QUIET_HOURS_START_HOUR < QUIET_HOURS_END_HOUR) {
    return hour >= QUIET_HOURS_START_HOUR && hour < QUIET_HOURS_END_HOUR;
  }
  return hour >= QUIET_HOURS_START_HOUR || hour < QUIET_HOURS_END_HOUR;
}

// Tarea 8.3: el modo stream retiene los avisos igual que el horario de no
// molestar -- se acumulan y se leen apenas termina (ver streamMode.ts). Lo
// mismo el modo juego (ver gameMode.ts).
export function shouldHoldAnnouncements(): boolean {
  return isQuietHours() || isStreamModeActive() || isGameModeActive();
}

File: Miku-Android/app/src/main/java/com/sebas/mikuai/data/QuietHours.kt
package com.sebas.mikuai.data

import java.util.Calendar

// Idea #20: franja horaria en la que ningún aviso automático (correo,
// Calendar, pendientes) habla solo -- se acumulan y se leen cuando termina.
// Mismo criterio que el lado desktop (lib/quietHours.ts): soporta que la
// franja cruce la medianoche además del caso simple (0 a 7) que se usa por
// defecto.
private const val QUIET_HOURS_START_HOUR = 0 // 00:00
private const val QUIET_HOURS_END_HOUR = 7 // 07:00

fun isQuietHours(): Boolean {
    val hour = Calendar.getInstance().get(Calendar.HOUR_OF_DAY)
    if (QUIET_HOURS_START_HOUR == QUIET_HOURS_END_HOUR) return false
    return if (QUIET_HOURS_START_HOUR < QUIET_HOURS_END_HOUR) {
        hour >= QUIET_HOURS_START_HOUR && hour < QUIET_HOURS_END_HOUR
    } else {
        hour >= QUIET_HOURS_START_HOUR || hour < QUIET_HOURS_END_HOUR
    }
}
